### PR TITLE
refactor: unify usage of types::Field in the prover crate

### PR DESCRIPTION
**Pull request by uttkarzsh**
Implements the changes described in issue https://github.com/NethermindEth/stellar-private-payments/issues/170

Summary of changes:
Structures now use `types::Field`  and use arkworks `Scalar` type only when needed for hashing. Also updated the Merkle Tree structures as well to use Field instead of Scalar.
Used standard circuit input encoding using `Field::to_le_bytes()`

- cargo test -p prover --tests  (all passed)
- cargo clippy passed
- cargo fmt applied

continued from pr https://github.com/NethermindEth/stellar-private-payments/pull/148

### CODE REVIEW

**Comment by maksimryndin** [comment]
hey @uttkarzsh ! ✋  Thanks a lot for the contribtuion!
Could we also please make sure https://github.com/NethermindEth/stellar-private-payments/blob/feature-unified-state/app/crates/core/prover/src/flows.rs#L31 and https://github.com/NethermindEth/stellar-private-payments/blob/feature-unified-state/app/crates/core/prover/src/flows.rs#L75 to use `types::Field` instead of plain `[u32; 32]`? where the field element is expected?

File: app/crates/platforms/web/src/workers/storage.rs [hunk 1]
```diff
@@ -636,6 +636,11 @@ fn build_pool_inputs(
             })?;
 
         let (path_elements, path_indices) = tree.proof_bytes(leaf_index)?;
```
Comment (maksimryndin) [inline app/crates/platforms/web/src/workers/storage.rs]: Maybe we should change `proof_bytes` signature to return Vec<Field> and Field instead of arrays (probably the method name should also be changed then)? wdyt @uttkarzsh ?

If that method used only here, then probably it makes sense?

File: app/crates/platforms/web/src/workers/prover.rs [hunk 2]
```diff
@@ -178,17 +178,11 @@ fn prove_from_artifacts(transact_artifacts: TransactArtifacts) -> Result<Prepare
         }
 
         let p = transact_artifacts.prepared.clone();
```
Comment (maksimryndin) [inline app/crates/platforms/web/src/workers/prover.rs]: A small improvement we can introduce here (I didn't do that initially) - probably, we don't need clone here - we can deconstruct the struct. And then only clone `ext_data` earlier.

File: app/crates/core/prover/src/sparse_merkle.rs [hunk 3]
```diff
@@ -403,24 +419,24 @@ impl WasmSparseMerkleTree {
...
         let find_result = self.inner.find(&key).map_err(|e| anyhow!(e))?;
 
         // Pad siblings to max_levels
         let mut siblings = find_result.siblings.clone();
         while siblings.len() < max_levels {
-            siblings.push(Scalar::from(0u64));
+            siblings.push(Field::ZERO);
         }
 
         Ok(WasmSMTProof {
```
Comment (maksimryndin) [inline app/crates/core/prover/src/sparse_merkle.rs]: Let's rename this struct to just `SMTProof`

File: app/crates/core/prover/src/merkle.rs [hunk 4]
```diff
@@ -6,44 +6,58 @@
...
-use types::Field as AppField;
+use crate::{crypto, serialization::scalar_to_bytes};
+use types::Field;
 
 // Re-export core merkle functions from circuits
 pub use circuits::core::merkle::{
     merkle_proof as merkle_proof_internal, merkle_root, poseidon2_compression,
 };
 
+fn field_to_scalar(field: Field) -> Scalar {
```
Comment (maksimryndin) [inline app/crates/core/prover/src/merkle.rs:19]: could we add a few simple unit tests here to check the conversion at edge cases? (e.g. at the modulo p - 1, at modulo p, at 0 etc)?

**Review by maksimryndin — COMMENTED** [review]
Looks good! 👍  thanks a lot @uttkarzsh !

some nit picks!

File: app/crates/platforms/web/src/workers/storage.rs (hunk 1 above)
Comment (uttkarzsh) [reply app/crates/platforms/web/src/workers/storage.rs]: Changed the call from proof_bytes() to proof() which already returns Vec and Field that we need. If we change the signature, we will be kind of duplicating what proof() already does for us. And proof_bytes is only left in the test cases at the moment, so i kept it as-is for now as a byte-level helper if needed in the future (maybe). If you’d like I can remove it entirely and update the test cases. :)

File: app/crates/platforms/web/src/workers/prover.rs (hunk 2 above)
Comment (uttkarzsh) [reply app/crates/platforms/web/src/workers/prover.rs]: I tried destructuring, but prepared is a PreparedTx (not PreparedTxPublic) and the field names differ (public_amount vs public_amount_field), so I kept the explicit mapping from PreparedTx → PreparedTxPublic to avoid type/field mismatches. I still removed the clone()   and now clone ext_data up front. cargo check -p web --all-features passes. lmk there is something else that you’d like me to do :)

File: app/crates/core/prover/src/sparse_merkle.rs (hunk 3 above)
Comment (uttkarzsh) [reply app/crates/core/prover/src/sparse_merkle.rs]: Done 👍

File: app/crates/core/prover/src/merkle.rs (hunk 4 above)
Comment (uttkarzsh) [reply app/crates/core/prover/src/merkle.rs:19]: Done. Tests now cover 0, 1, modulus - 1, and modulus (reduces to 0).

File: app/crates/platforms/web/src/workers/storage.rs (hunk 1 above)
Comment (maksimryndin) [reply app/crates/platforms/web/src/workers/storage.rs]: let's remove proof_bytes then if don't use it anywhere else except tests?

Comment (uttkarzsh) [reply app/crates/platforms/web/src/workers/storage.rs]: Done :)

**Comment by maksimryndin** [comment]
Hey @uttkarzsh! ✋ 

Sorry, seems GH closed automatically - could you pls rebase you PR on the latest main?
